src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -129,6 +129,24 @@ private struct Entry {
             }
         }
 
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> enumerable):
+            this(enumerable, EqualityComparer<TKey>.Default) { }
+
+        public Dictionary(IEnumerable<KeyValuePair<TKey, TValue>> enumerable, IEqualityComparer<TKey> comparer):
+            this(comparer)
+        {
+            if( enumerable == null) {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.enumerable);
```

Comment:
I used ThrowHelper here because it was used all around in Dictionary.cs. But have a question. Is there any win over:

`throw ArgumentNullException(nameof(enumerable)`

?